scanpy/readwrite.py

```diff
@@ -351,18 +365,37 @@ def read_visium(
                 else:
                     raise OSError(f"Could not find '{f}'")
 
-        adata.uns['images'] = dict()
+        adata.uns["spatial"][library_id]['images'] = dict()
         for res in ['hires', 'lowres']:
             try:
-                adata.uns['images'][res] = imread(str(files[f'{res}_image']))
+                adata.uns["spatial"][library_id]['images'][res] = imread(
+                    str(files[f'{res}_image'])
+                )
             except Exception:
```

Comment:
This should be more specific. Also, it's pretty bad if you tried to read in a spatial dataset, and it failed right? Why shouldn't this just error?

Reply:
Technically no, it could still be the case that there is no image but spatial coordinates. It would still make sense to read it in as visium data, excetp the image entry would be empty.

Reply:
So it's possible to run the pipeline without images? How do the images get to the pipeline?

I'm wondering if the number and names of the image files might be variable.

Reply:
It looks to me like these images [have to exist and have to have these names](https://support.10xgenomics.com/spatial-gene-expression/software/pipelines/latest/output/images).

Reply:
Yeah that's true, I thought it's just better to give such flexibility for whatever reason (e.g. shared data with missing stuff) but can change to raise an error

Reply:
I think this should raise an error. Since this function is specifically for reading visium experiments it reduces complexity a lot to just assume they ran the visium pipeline. Anything more homebrew will also take a more homebrew solution for reading in.

Reply:
Done

Reply:
One problem is that I had to add the `hires` image to the `_data` folder for test. That image is ~5 Mb. This is potentially a problem considering that now it accounts for half of the `_data` size

Reply:
This example data has been subset, right? Could you share the script used to generate it?

Could the size of the image be reduced by removing unimportant data? I'm thinking setting unused areas of the image to a solid color, so they'll compress away.

Reply:
I thought and in fact it doesn't really, I just pushed an image with same resolution but not colors. Should be fine for testing.

yes, the dataset is a subset, and it's been created with a combination of [dropletutils](https://github.com/MarioniLab/DropletUtils) functions and h5py. This was the easiest way I found to subset the dataset and keep attributes in place.